### PR TITLE
Adding stencil redux docs

### PR DESCRIPTION
**Pull request by corysmc**
Stencil redux docs with a demo walkthrough

### CODE REVIEW

File: src/docs/guides/redux.md [hunk 1]
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
...
+export class MyApp {
+  @State()
+  name: MyAppState["user"]["name"];
+
+  @Prop({ context: "store" })
+  store: Store;
+
+  async componentWillLoad() {
+    this.store.setStore(configureStore({}));
+    this.store.mapStateToProps(this, (state: MyAppState) => {
```
Comment (bitflower) [inline src/docs/guides/redux.md]: We should make this:
```
this.storeUnsubscribe = this.store.mapStateToProps(this, (state: MyAppState) => {
```

File: src/docs/guides/redux.md [hunk 2]
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
...
+
+## 5. Initilize Store
+
+Initialize the store from within your root component. The store can then be accessed from within other components (see step 5).
+
+```tsx
+// /src/components/app-root/app-root.tsx
+import "@stencil/redux";
+import { Component, Prop, State, h } from "@stencil/core";
+import { Store } from "@stencil/redux";
```
Comment (bitflower) [inline src/docs/guides/redux.md]: Add `Unsubscribe` to imports.

File: src/docs/guides/redux.md [hunk 3]
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
...
+  tag: "app-root",
+  styleUrl: "app-root.css"
+})
+export class MyApp {
+  @State()
+  name: MyAppState["user"]["name"];
+
+  @Prop({ context: "store" })
+  store: Store;
+
```
Comment (bitflower) [inline src/docs/guides/redux.md]: Add `storeUnsubscribe: Unsubscribe;`

File: src/docs/guides/redux.md [hunk 4]
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
...
+    this.store.mapStateToProps(this, (state: MyAppState) => {
+      const {
+        user: { name }
+      } = state;
+      return {
+        name
+      };
+    });
+  }
+
```
Comment (bitflower) [inline src/docs/guides/redux.md]: Add:
```
  protected componentDidUnload() {
    if (this.storeUnsubscribe) {
      this.storeUnsubscribe();
    }
  }
```

File: src/docs/guides/redux.md [hunk 5]
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
 contributors:
   - jthoms1
   - adamdbradley
+  - corysmc
 ---
```
Comment (bitflower) [inline src/docs/guides/redux.md]: If you like my changes add `bitflower` ;-)

**Comment by corysmc** [comment]
This is awesome, thanks @bitflower  I seriously just ran into this issue, had no idea what was causing it. Just updated the docs, and added you as a contributor

**Comment by adamdbradley** [comment]
🚀

**Pull request merged**